libutils/json.c

```diff
@@ -296,31 +298,24 @@ static int JsonObjectCompare(
     int ret = JsonLength(a) - JsonLength(b);
     if (ret != 0)
     {
+        Log(LOG_LEVEL_DEBUG, "JsonObjectCompare() fails, length differs by %d", ret);
         return ret;
     }
 
-    JsonIterator iter_a = JsonIteratorInit(a);
-    JsonIterator iter_b = JsonIteratorInit(a);
+    JsonIterator iter = JsonIteratorInit(a);
 
-    for (size_t i = 0; i < JsonLength(a); i++)
+    while (JsonIteratorHasMore(&iter))
     {
-        const JsonElement *child_a = JsonIteratorNextValue(&iter_a);
-        const JsonElement *child_b = JsonIteratorNextValue(&iter_b);
-
-        const char *const key_a = JsonIteratorCurrentKey(&iter_a);
-        const char *const key_b = JsonIteratorCurrentKey(&iter_b);
-
-        ret = strcmp(key_a, key_b);
-        if (ret != 0)
-        {
-            return ret;
-        }
+        const char *const key = JsonIteratorNextKey(&iter);
+        const JsonElement *const child_a = JsonObjectGet(a, key);
+        const JsonElement *const child_b = JsonObjectGet(b, key);
 
-        ret = JsonCompare(child_a, child_b);
-        if (ret != 0)
+        if (child_b == NULL || JsonCompare(child_a, child_b))
         {
-            return ret;
+            Log(LOG_LEVEL_DEBUG, "JsonObjectCompare() fails for key '%s', children not equal", key);
+            return 1;
```

Comment:
I feel like just to be sure saying 
```suggestion
            return 1; // not equal like strcmp()
```